DEVELOPER.md

```diff
@@ -0,0 +1,154 @@
+# Developing Siemens Element
+
+## Prerequisites
+
+- [Git](https://git-scm.com/) to clone the repository
+- [Git LFS](https://git-lfs.github.com/) to fetch test assets
+- [NodeJS](https://nodejs.org/) Version as specified in the [package.json](package.json)
+
+To run the documentation, you will also need:
+
+- [Python](https://www.python.org/) Version as specified in [pyproject.toml](pyproject.toml)
+- [UV](https://uv.readthedocs.io/en/latest/) to manage Python virtual environments
+
+## Installation
+
+Setting up the development environment:
+
+```shell
+git clone git@github.com:siemens/element.git
+cd element
+git lfs install
+npm install
+```
+
+## Building
+
+We use path mappings within the repository.
+For development builds, they typically point to the sources,
+so building submodules is not necessary.
+
+**IMPORTANT**: Always build the translation layer. It is not consumed form the sources.
+
+To build modules use the available npm scripts:
+
+```shell
+# Build all modules
+npm run build:all
+
+# Build a specific module. See package.json for available modules.
+npm run <module-name>:build
+
+# e.g. Build the translation layer
+npm run translate:build
+```
+
+## Running the example application
+
+As for other builds, the translation layer must be built before running the application.
+Then you can example application with the corresponding npm script:
+
+```shell
+# Run the example application
+npm run start
+```
+
+## Linting and formatting
+
+We use [ESLint](https://eslint.org/) and [Stylelint](https://stylelint.io/) for linting and [Prettier](https://prettier.io/) for formatting.
+Use the following npm scripts to lint and format the code:
+
+```shell
+# Format all modules
+npm run format
+
+# Lint all modules
+npm run lint:ng
+
+# Lint styles
+npm run lint:scss
+```
+
+## Unit Tests
+
+Runt the unit tests using the corresponding npm script:
+
+```shell
+npm run <module-name>:test
+```
+
+## E2E Tests
+
+Our E2E tests are built with [Playwright](https://playwright.dev/).
+To ensure reliable screenshots, those tests must always be run in a docker container.
+
+To run the E2E tests, use the following commands:
+
+```shell
+# On the extra parameter can be omitted
+npm run start --  --host 127.0.0.1
+
+# Run the E2E tests on another terminal
+./e2e-local.sh
+
+# To update the e2e tests append: update or update-all
+./e2e-local.sh update
+
+# To only update a specific test, append the file name. Glob patterns are supported.
+./e2e-local.sh update <test-file-name>
+
+# To run a specific static test, use an environment variable and restrict the to static specs:
+PLAYWRIGHT_staticTest=buttons/buttons ./e2e-local.sh run static
```

Comment:
Shall we also mention here that multiple static examples can be provided with the `:` delimeter? One only knows about this when looking up the code...

Reply:
done